nautobot/docs/user-guides/git-data-source.md

```diff
@@ -0,0 +1,257 @@
+# Git as a Data Source
+
+Using Git&trade; as a Data Source feature was developed to provide the ability to populate existing data, templates, scripts, and much more into Nautobot; while leveraging the benefits that GitHub already provides including issue tracking, discussions, pipelines, and approvals.
+
+For more technical details on how to use this feature, please see the documentation on [Git Repositories](https://nautobot.readthedocs.io/en/latest/models/extras/gitrepository/).
+
+
+## Supported Providers
+The feature uses the concept of a `provides` field to map a repository to a use case. A list of the supported options is provided below.
+
+|Name|Summary|Related Plugin|
+|:--|:--|:--|
+|Export Templates|Nautobot allows users to define custom templates that can be used when exporting objects.|Core Functionality|
+|Jobs|Jobs are a way for users to execute custom logic on demand from within the Nautobot UI. Jobs can interact directly with Nautobot data to accomplish various data creation, modification, and validation tasks.|Core Functionality|
+|Config Contexts|Config contexts can be used to provide additional data that you can't natively store in Nautobot.|Core Functionality|
+|Backup Configs|Backup configuration data.|[Golden Config](https://github.com/nautobot/nautobot-plugin-golden-config)|
+|Intended Configs|Stores the intended configurations, this grabs Nautobot data and runs through Jinja Templates.|[Golden Config](https://github.com/nautobot/nautobot-plugin-golden-config)|
+|Jinja Templates|Repository that holds Jinja templates to be used to generate intended configs.|[Golden Config](https://github.com/nautobot/nautobot-plugin-golden-config)|
+
+## Repository Details
+
+Parameters:
+|Field|Explanation|
+|:--|:--|
+|Name|User friendly name for the repo.|
+|Slug|Auto-generated based on the `name` provided.|
+|Remote URL|The URL pointing to the Git repo. Current git url usage is limited to `http` or `https`.|
+|Branch|The branch in the Git repo to use. Defaults to `main`.|
+|Token|The token is a personal access token for the `username` provided.  For more information on generating a personal access token. [GitHub Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
```

Comment:
We should be clear that this isn't GitHub-exclusive - much like the `nautobot/docs/models/extras/gitrepository.md`, this should probably include references to GitLab and Bitbucket as well?